cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreImpl.java

```diff
@@ -1504,19 +1506,21 @@ public int compare(String r1, String r2) {
      * @param revHistory revision history to insert, it includes all revisions (include the revision of the DocumentRevision
      *                   as well) sorted in ascending order.
      */
-    private long doForceInsertNewDocumentWithHistory(SQLDatabase db,
-                                                     DocumentRevision rev,
-                                                     List<String> revHistory)
-            throws AttachmentException{
+    private InsertRevisionCallable.Result doForceInsertNewDocumentWithHistory(SQLDatabase db,
+                                                                              DocumentRevision rev,
+                                                                              List<String> revHistory)
+            throws AttachmentException, SQLException {
         logger.entering("DocumentRevision",
                 "doForceInsertNewDocumentWithHistory()",
                 new Object[]{rev, revHistory});
 
         long docNumericID = insertDocumentID(db,rev.getId());
         long parentSequence = 0L;
         for (int i = 0; i < revHistory.size() - 1; i++) {
+            //TODO: We could just continue here, if we have a dup we can just ignore
```

Comment:
We should probably have a warning in the log here - it might help us in the future to know that customers are running multiple replications in parallel etc.